packages/contracts-bedrock/test/L2/L2ToL2CrossDomainMessenger.t.sol

```diff
@@ -248,6 +299,48 @@ contract L2ToL2CrossDomainMessengerTest is Test {
         );
     }
 
+    /// @dev Tests that the `relayMessage` function reverts if called by an address other than the specified entrypoint.
+    function testFuzz_relayMessage_wrongEntrypoint_reverts(
+        uint256 _source,
+        uint256 _nonce,
+        address _sender,
+        bytes calldata _message,
+        address _entrypoint,
+        uint256 _value,
+        uint256 _blockNum,
+        uint256 _logIndex,
+        uint256 _time
+    )
+        external
+    {
+        address _target = makeAddr("target");
+
+        // Ensure entrypoint is not address(0) or the L2ToL2CrossDomainMessenger contract
+        vm.assume(_entrypoint != Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER && _entrypoint != address(0));
+
+        // Construct the SentMessage payload & identifier
+        Identifier memory id =
+            Identifier(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER, _blockNum, _logIndex, _time, _source);
+        bytes memory sentMessage = abi.encodePacked(
+            abi.encode(L2ToL2CrossDomainMessenger.SentMessage.selector, block.chainid, _target, _nonce), // topics
+            abi.encode(_sender, _message, address(_entrypoint)) // data
+        );
+
+        // Ensure the CrossL2Inbox validates this message
+        vm.mockCall({
+            callee: Predeploys.CROSS_L2_INBOX,
+            data: abi.encodeCall(CrossL2Inbox.validateMessage, (id, keccak256(sentMessage))),
+            returnData: ""
+        });
+
+        // Expect a revert with the InvalidEntrypoint selector
+        vm.expectRevert(InvalidEntrypoint.selector);
+
+        // Call
+        hoax(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER, _value);
```

Comment:
We could fuzz a caller different than `_entrypoint` here as well to cover more possible scenarios on the test

Reply:
Thought about it but the ideas was to keep the tests style.